tools/pipeline_perf_test/orchestrator/orchestrator.py

```diff
@@ -1,17 +1,102 @@
 import argparse
 import os
-import time
-import requests
+import re
 import subprocess
-import json
+import threading
+import time
 from datetime import datetime
-from typing import Optional, Dict, Any, List, Union, Tuple
+from typing import Optional, Dict, List, Tuple
+
+import requests
+
+
+class ProcessStatsAggregation:
+    """Class for aggregation of resource stats"""
+    min: float
+    max: float
+    total: float
+    samples: int
+
+    def __init__(self):
+        self.min = None
+        self.max = None
+        self.total = 0.0
+        self.samples = 0
+
+    def add_sample(self, sample: float):
```

Comment:
https://github.com/open-telemetry/otel-arrow/issues/444
In a future PR, we could switch to using Otel Python Metrics API!

Reply:
big plus 1!